core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/SeleniumGridImpl.java

```diff
@@ -0,0 +1,118 @@
+        updateQueueOfBrowsers();
+    }
+
+    protected void updateQueueOfBrowsers() {
+        timerTask =
+                new TimerTask() {
+                    @Override
+                    public void run() {
+                        try {
+                            List list = getAllNodesList();
+                            LOG.info("Blocking Queue size: " + driversQueue.size());
+                            while (getSessionsCount(list) > 0) {
+                                int size = list.size();
+                                // Check if the queue size is more than the actual
+                                // no of browsers allowed per worker
+                                // means crawler services are idle because all drivers are in queue
+                                if (driversQueue.size() >= (size / noOfWorkers)) {
+                                    SeleniumGridProtocol.Holder holder = driversQueue.take();
+                                    long totalTime = System.currentTimeMillis() - holder.getTime();
+                                    // clear the queue if the total time spent in the queue is more
+                                    // than 4.5 minutes
+                                    // As after 5 mintues the driver would be idle and will throw
+                                    // exception
+                                    // while we try to use that driver for fetching
+                                    if (totalTime > 1000 * 4.5 * 60) {
+                                        driversQueue.clear();
+                                    }
+                                }
+                                // so that browsers get equally divided among workers
+                                if (driversQueue.size() <= size / noOfWorkers) {
+                                    RemoteWebDriver driver = getDriverFromNode();
+                                    if (driver != null) {
+                                        driversQueue.put(
+                                                new SeleniumGridProtocol.Holder(
+                                                        driver, System.currentTimeMillis()));
+                                        LOG.info(
+                                                "Placed driver in blocking queue: "
+                                                        + driversQueue.size());
+                                    }
+                                }
+                                list = getAllNodesList();
+                            }
+                        } catch (Exception e) {
+                            LOG.error(
+                                    "Exception while running task for adding driver to the queue",
+                                    e);
+                        }
+                    }
+                };
+        // update the queue every 5 minutes
+        updateQueue.schedule(timerTask, 0 * 60 * 1000, 5 * 60 * 1000);
```

Comment:
The first integer value will always be 0.

Can we configure that value?

Reply:
sure

Reply:
![image](https://github.com/DigitalPebble/storm-crawler/assets/32441552/2e546fc3-1e54-45ec-8fa2-b5df61e57839)

I am naming it 'start.queuing.delay' suggest a property name if you want something different

Reply:
The second value (5min) should most likely also be configurable as it depends on the timeout configured for the grid (default = 5min)

Reply:
made it selenium.grid.delay